app/utils/getCacheAnalysis.test.ts

```diff
@@ -91,3 +359,305 @@ describe('getTimeToLive', () => {
     expect(getTimeToLive(age, date, expiresAt, maxAge, now)).toBeUndefined()
   })
 })
+
+// Test helper functions for internal functions since they're not exported
+// We'll test them through the main functions where possible, but these are edge cases
+
+describe('internal helper functions', () => {
+  describe('fixDuplicatedCdnNodes (through getServedBy)', () => {
+    it('removes duplicate CDN nodes from Debug-X-BB-Host-Id header', () => {
+      const headers = {
+        'Cache-Status': '"Netlify Edge"; hit',
+        'Debug-X-BB-Host-Id': 'node1.example.com, node1.example.com, node2.example.com',
+      }
+      const now = Date.now()
+
+      const result = getCacheAnalysis(headers, now)
+
+      expect(result.servedBy.cdnNodes).toBe('node1.example.com, node2.example.com')
+    })
+
+    it('preserves unique CDN nodes in Debug-X-BB-Host-Id header', () => {
+      const headers = {
+        'Cache-Status': '"Netlify Edge"; hit',
+        'Debug-X-BB-Host-Id': 'node1.example.com, node2.example.com, node3.example.com',
+      }
+      const now = Date.now()
+
+      const result = getCacheAnalysis(headers, now)
+
+      expect(result.servedBy.cdnNodes).toBe('node1.example.com, node2.example.com, node3.example.com')
+    })
+
+    it('handles single CDN node without duplicates', () => {
+      const headers = {
+        'Cache-Status': '"Netlify Edge"; hit',
+        'Debug-X-BB-Host-Id': 'node1.example.com',
+      }
+      const now = Date.now()
+
+      const result = getCacheAnalysis(headers, now)
+
+      expect(result.servedBy.cdnNodes).toBe('node1.example.com')
+    })
+
+    it('uses fallback when Debug-X-BB-Host-Id header is missing', () => {
+      const headers = {
+        'Cache-Status': '"Netlify Edge"; hit',
+      }
+      const now = Date.now()
+
+      const result = getCacheAnalysis(headers, now)
+
+      expect(result.servedBy.cdnNodes).toBe('unknown CDN node')
+    })
+  })
+
+  describe('TTL precedence order (through parseCacheControl)', () => {
```

Comment:
export it and unit test it instead! to be clear... not here. add a test file for `app/utils/cache-control.ts`